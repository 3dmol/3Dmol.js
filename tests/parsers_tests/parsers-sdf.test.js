
global.$ = require("jquery");
global.URL.createObjectURL = function() {};
let $3Dmol = require("../../build/3Dmol.js");
const fs = require('fs');
const path = require("path");

describe('Function SDF | parseV2000 |', ()=>{
    const data = fs.readFileSync(path.resolve(__dirname, '../test_structs/aromaticsdf.sdf'), 'utf-8')
    // undefined keepH
    let atoms = $3Dmol.Parsers.SDF(data, {});
    let atomCount = atoms[0].length;
    
    test("Atoms is empty when input data is not greater than 3 lines",()=>{
        let test_data1 = "line1\nline2\nline3";
        let test_atoms1 = $3Dmol.Parsers.SDF(test_data1, {});
        expect(test_atoms1).toEqual([[]]);
    });

    test("Atoms is empty when length of line 4 of input data is less than 38",()=>{
        let test_data2 = "line1\nline2\nline3\nlessthan38";
        let test_atoms2 = $3Dmol.Parsers.SDF(test_data2, {});
        expect(test_atoms2).toEqual([[]]);
    });

    test("Atoms is empty when atomCount is not a number", ()=>{
        let test_data3 = "line1\nline2\nline3\natomCount";
        let test_atoms3 = $3Dmol.Parsers.SDF(test_data3, {});
        expect(test_atoms3).toEqual([[]]);
    });

    test("Atoms is empty when atomCount is not greater than 0", ()=>{
        let test_data4 = "line1\nline2\nline3\n -1";
        let test_atoms4 = $3Dmol.Parsers.SDF(test_data4, {});
        expect(test_atoms4).toEqual([[]]);
    });

    test("Atoms is empty when the number of lines of input is less than 4 + atomCount + bondCount", ()=>{
        let test_data5 = "line1\nline2\nline3\n  1 0 ";// atomCount=0, bondCount=1
        let test_atoms5 = $3Dmol.Parsers.SDF(test_data5, {});
        expect(test_atoms5).toEqual([[]]);
    });

    test("Length of atoms is 1", ()=>{
        expect(atoms.length).toBe(1); 
    });
    
    test("Atoms should match the snapshot", ()=>{
        expect(atoms).toMatchSnapshot();
    });

});

// keepH = false; noH = true;
describe('Function SDF | parseV2000 | options: keepH |', ()=>{
    const data = fs.readFileSync(path.resolve(__dirname, '../test_structs/aromaticsdf.sdf'), 'utf-8')
    let atoms = $3Dmol.Parsers.SDF(data, {keepH:false});

    test("Atoms should match the snapshot when keepH is false", ()=>{
        expect(atoms).toMatchSnapshot();
    });
    
});

// multimodel
describe('Function SDF | parseV2000 | options: multimodel, onemol |', ()=>{
    const data = fs.readFileSync(path.resolve(__dirname, '../test_structs/aromaticsdf.sdf'), 'utf-8')

    test("Atoms should match the snapshot when multimodel is true and onemol is false/undefinedd", ()=>{
        let atoms = $3Dmol.Parsers.SDF(data, {multimodel:true, onemol:false}); 
        expect(atoms).toMatchSnapshot();
    });

    test("Atoms should match the snapshot when multimodel is true and onemol is true", ()=>{
        let atoms = $3Dmol.Parsers.SDF(data, {multimodel:true, onemol:true}); 
        expect(atoms).toMatchSnapshot();
    });

});
