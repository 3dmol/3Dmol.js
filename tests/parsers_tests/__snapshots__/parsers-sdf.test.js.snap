// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Function SDF | parseV2000 | Atoms should match the snapshot 1`] = `
[
  [
    {
      "atom": "C",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        4,
        10,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 0,
      "properties": {},
      "serial": 0,
      "x": -9.856,
      "y": 4.168,
      "z": 6.715,
    },
    {
      "atom": "C",
      "bondOrder": [
        1,
      ],
      "bonds": [
        2,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 1,
      "properties": {},
      "serial": 1,
      "x": -12.153,
      "y": 2.39,
      "z": 5.307,
    },
    {
      "atom": "C",
      "bondOrder": [
        1,
        4,
        4,
      ],
      "bonds": [
        1,
        4,
        5,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 2,
      "properties": {},
      "serial": 2,
      "x": -11.98,
      "y": 3.867,
      "z": 5.218,
    },
    {
      "atom": "C",
      "bondOrder": [
        4,
        4,
        1,
      ],
      "bonds": [
        4,
        6,
        7,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 3,
      "properties": {},
      "serial": 3,
      "x": -11.355,
      "y": 5.923,
      "z": 5.494,
    },
    {
      "atom": "C",
      "bondOrder": [
        1,
        4,
        4,
      ],
      "bonds": [
        0,
        2,
        3,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 4,
      "properties": {},
      "serial": 4,
      "x": -10.967,
      "y": 4.626,
      "z": 5.788,
    },
    {
      "atom": "N",
      "bondOrder": [
        4,
        4,
      ],
      "bonds": [
        2,
        6,
      ],
      "elem": "N",
      "hetflag": true,
      "index": 5,
      "properties": {},
      "serial": 5,
      "x": -12.865,
      "y": 4.669,
      "z": 4.596,
    },
    {
      "atom": "N",
      "bondOrder": [
        4,
        4,
        1,
      ],
      "bonds": [
        3,
        5,
        8,
      ],
      "elem": "N",
      "hetflag": true,
      "index": 6,
      "properties": {},
      "serial": 6,
      "x": -12.483,
      "y": 5.926,
      "z": 4.762,
    },
    {
      "atom": "O",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        3,
        9,
      ],
      "elem": "O",
      "hetflag": true,
      "index": 7,
      "properties": {},
      "serial": 7,
      "x": -10.714,
      "y": 7.084,
      "z": 5.905,
    },
    {
      "atom": "H",
      "bondOrder": [
        1,
      ],
      "bonds": [
        6,
      ],
      "elem": "H",
      "hetflag": true,
      "index": 8,
      "properties": {},
      "serial": 8,
      "x": -12.943,
      "y": 6.722,
      "z": 4.411,
    },
    {
      "atom": "H",
      "bondOrder": [
        1,
      ],
      "bonds": [
        7,
      ],
      "elem": "H",
      "hetflag": true,
      "index": 9,
      "properties": {},
      "serial": 9,
      "x": -9.929,
      "y": 7.031,
      "z": 6.415,
    },
    {
      "atom": "C",
      "bondOrder": [
        1,
        4,
        4,
      ],
      "bonds": [
        0,
        11,
        12,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 10,
      "properties": {},
      "serial": 10,
      "x": -9.716,
      "y": 4.99,
      "z": 7.966,
    },
    {
      "atom": "C",
      "bondOrder": [
        4,
        4,
        1,
      ],
      "bonds": [
        10,
        14,
        16,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 11,
      "properties": {},
      "serial": 11,
      "x": -8.698,
      "y": 5.843,
      "z": 8.337,
    },
    {
      "atom": "C",
      "bondOrder": [
        4,
        1,
        4,
      ],
      "bonds": [
        10,
        13,
        15,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 12,
      "properties": {},
      "serial": 12,
      "x": -10.564,
      "y": 5.007,
      "z": 9.052,
    },
    {
      "atom": "C",
      "bondOrder": [
        1,
      ],
      "bonds": [
        12,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 13,
      "properties": {},
      "serial": 13,
      "x": -11.829,
      "y": 4.236,
      "z": 9.194,
    },
    {
      "atom": "N",
      "bondOrder": [
        4,
        4,
        1,
      ],
      "bonds": [
        11,
        15,
        17,
      ],
      "elem": "N",
      "hetflag": true,
      "index": 14,
      "properties": {},
      "serial": 14,
      "x": -8.933,
      "y": 6.343,
      "z": 9.563,
    },
    {
      "atom": "N",
      "bondOrder": [
        4,
        4,
      ],
      "bonds": [
        12,
        14,
      ],
      "elem": "N",
      "hetflag": true,
      "index": 15,
      "properties": {},
      "serial": 15,
      "x": -10.075,
      "y": 5.83,
      "z": 10.001,
    },
    {
      "atom": "O",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        11,
        18,
      ],
      "elem": "O",
      "hetflag": true,
      "index": 16,
      "properties": {},
      "serial": 16,
      "x": -7.582,
      "y": 6.162,
      "z": 7.573,
    },
    {
      "atom": "H",
      "bondOrder": [
        1,
      ],
      "bonds": [
        14,
      ],
      "elem": "H",
      "hetflag": true,
      "index": 17,
      "properties": {},
      "serial": 17,
      "x": -8.359,
      "y": 6.978,
      "z": 10.05,
    },
    {
      "atom": "H",
      "bondOrder": [
        1,
      ],
      "bonds": [
        16,
      ],
      "elem": "H",
      "hetflag": true,
      "index": 18,
      "properties": {},
      "serial": 18,
      "x": -7.471,
      "y": 5.782,
      "z": 6.724,
    },
  ],
]
`;

exports[`Function SDF | parseV2000 | options: keepH | Atoms should match the snapshot when keepH is false 1`] = `
[
  [
    {
      "atom": "C",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        4,
        8,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 0,
      "properties": {},
      "serial": 0,
      "x": -9.856,
      "y": 4.168,
      "z": 6.715,
    },
    {
      "atom": "C",
      "bondOrder": [
        1,
      ],
      "bonds": [
        2,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 1,
      "properties": {},
      "serial": 1,
      "x": -12.153,
      "y": 2.39,
      "z": 5.307,
    },
    {
      "atom": "C",
      "bondOrder": [
        1,
        4,
        4,
      ],
      "bonds": [
        1,
        4,
        5,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 2,
      "properties": {},
      "serial": 2,
      "x": -11.98,
      "y": 3.867,
      "z": 5.218,
    },
    {
      "atom": "C",
      "bondOrder": [
        4,
        4,
        1,
      ],
      "bonds": [
        4,
        6,
        7,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 3,
      "properties": {},
      "serial": 3,
      "x": -11.355,
      "y": 5.923,
      "z": 5.494,
    },
    {
      "atom": "C",
      "bondOrder": [
        1,
        4,
        4,
      ],
      "bonds": [
        0,
        2,
        3,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 4,
      "properties": {},
      "serial": 4,
      "x": -10.967,
      "y": 4.626,
      "z": 5.788,
    },
    {
      "atom": "N",
      "bondOrder": [
        4,
        4,
      ],
      "bonds": [
        2,
        6,
      ],
      "elem": "N",
      "hetflag": true,
      "index": 5,
      "properties": {},
      "serial": 5,
      "x": -12.865,
      "y": 4.669,
      "z": 4.596,
    },
    {
      "atom": "N",
      "bondOrder": [
        4,
        4,
      ],
      "bonds": [
        3,
        5,
      ],
      "elem": "N",
      "hetflag": true,
      "index": 6,
      "properties": {},
      "serial": 6,
      "x": -12.483,
      "y": 5.926,
      "z": 4.762,
    },
    {
      "atom": "O",
      "bondOrder": [
        1,
      ],
      "bonds": [
        3,
      ],
      "elem": "O",
      "hetflag": true,
      "index": 7,
      "properties": {},
      "serial": 7,
      "x": -10.714,
      "y": 7.084,
      "z": 5.905,
    },
    {
      "atom": "C",
      "bondOrder": [
        1,
        4,
        4,
      ],
      "bonds": [
        0,
        9,
        10,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 8,
      "properties": {},
      "serial": 10,
      "x": -9.716,
      "y": 4.99,
      "z": 7.966,
    },
    {
      "atom": "C",
      "bondOrder": [
        4,
        4,
        1,
      ],
      "bonds": [
        8,
        12,
        14,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 9,
      "properties": {},
      "serial": 11,
      "x": -8.698,
      "y": 5.843,
      "z": 8.337,
    },
    {
      "atom": "C",
      "bondOrder": [
        4,
        1,
        4,
      ],
      "bonds": [
        8,
        11,
        13,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 10,
      "properties": {},
      "serial": 12,
      "x": -10.564,
      "y": 5.007,
      "z": 9.052,
    },
    {
      "atom": "C",
      "bondOrder": [
        1,
      ],
      "bonds": [
        10,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 11,
      "properties": {},
      "serial": 13,
      "x": -11.829,
      "y": 4.236,
      "z": 9.194,
    },
    {
      "atom": "N",
      "bondOrder": [
        4,
        4,
      ],
      "bonds": [
        9,
        13,
      ],
      "elem": "N",
      "hetflag": true,
      "index": 12,
      "properties": {},
      "serial": 14,
      "x": -8.933,
      "y": 6.343,
      "z": 9.563,
    },
    {
      "atom": "N",
      "bondOrder": [
        4,
        4,
      ],
      "bonds": [
        10,
        12,
      ],
      "elem": "N",
      "hetflag": true,
      "index": 13,
      "properties": {},
      "serial": 15,
      "x": -10.075,
      "y": 5.83,
      "z": 10.001,
    },
    {
      "atom": "O",
      "bondOrder": [
        1,
      ],
      "bonds": [
        9,
      ],
      "elem": "O",
      "hetflag": true,
      "index": 14,
      "properties": {},
      "serial": 16,
      "x": -7.582,
      "y": 6.162,
      "z": 7.573,
    },
  ],
]
`;

exports[`Function SDF | parseV2000 | options: multimodel, onemol | Atoms should match the snapshot when multimodel is true and onemol is false/undefinedd 1`] = `
[
  [
    {
      "atom": "C",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        4,
        10,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 0,
      "properties": {},
      "serial": 0,
      "x": -9.856,
      "y": 4.168,
      "z": 6.715,
    },
    {
      "atom": "C",
      "bondOrder": [
        1,
      ],
      "bonds": [
        2,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 1,
      "properties": {},
      "serial": 1,
      "x": -12.153,
      "y": 2.39,
      "z": 5.307,
    },
    {
      "atom": "C",
      "bondOrder": [
        1,
        4,
        4,
      ],
      "bonds": [
        1,
        4,
        5,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 2,
      "properties": {},
      "serial": 2,
      "x": -11.98,
      "y": 3.867,
      "z": 5.218,
    },
    {
      "atom": "C",
      "bondOrder": [
        4,
        4,
        1,
      ],
      "bonds": [
        4,
        6,
        7,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 3,
      "properties": {},
      "serial": 3,
      "x": -11.355,
      "y": 5.923,
      "z": 5.494,
    },
    {
      "atom": "C",
      "bondOrder": [
        1,
        4,
        4,
      ],
      "bonds": [
        0,
        2,
        3,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 4,
      "properties": {},
      "serial": 4,
      "x": -10.967,
      "y": 4.626,
      "z": 5.788,
    },
    {
      "atom": "N",
      "bondOrder": [
        4,
        4,
      ],
      "bonds": [
        2,
        6,
      ],
      "elem": "N",
      "hetflag": true,
      "index": 5,
      "properties": {},
      "serial": 5,
      "x": -12.865,
      "y": 4.669,
      "z": 4.596,
    },
    {
      "atom": "N",
      "bondOrder": [
        4,
        4,
        1,
      ],
      "bonds": [
        3,
        5,
        8,
      ],
      "elem": "N",
      "hetflag": true,
      "index": 6,
      "properties": {},
      "serial": 6,
      "x": -12.483,
      "y": 5.926,
      "z": 4.762,
    },
    {
      "atom": "O",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        3,
        9,
      ],
      "elem": "O",
      "hetflag": true,
      "index": 7,
      "properties": {},
      "serial": 7,
      "x": -10.714,
      "y": 7.084,
      "z": 5.905,
    },
    {
      "atom": "H",
      "bondOrder": [
        1,
      ],
      "bonds": [
        6,
      ],
      "elem": "H",
      "hetflag": true,
      "index": 8,
      "properties": {},
      "serial": 8,
      "x": -12.943,
      "y": 6.722,
      "z": 4.411,
    },
    {
      "atom": "H",
      "bondOrder": [
        1,
      ],
      "bonds": [
        7,
      ],
      "elem": "H",
      "hetflag": true,
      "index": 9,
      "properties": {},
      "serial": 9,
      "x": -9.929,
      "y": 7.031,
      "z": 6.415,
    },
    {
      "atom": "C",
      "bondOrder": [
        1,
        4,
        4,
      ],
      "bonds": [
        0,
        11,
        12,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 10,
      "properties": {},
      "serial": 10,
      "x": -9.716,
      "y": 4.99,
      "z": 7.966,
    },
    {
      "atom": "C",
      "bondOrder": [
        4,
        4,
        1,
      ],
      "bonds": [
        10,
        14,
        16,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 11,
      "properties": {},
      "serial": 11,
      "x": -8.698,
      "y": 5.843,
      "z": 8.337,
    },
    {
      "atom": "C",
      "bondOrder": [
        4,
        1,
        4,
      ],
      "bonds": [
        10,
        13,
        15,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 12,
      "properties": {},
      "serial": 12,
      "x": -10.564,
      "y": 5.007,
      "z": 9.052,
    },
    {
      "atom": "C",
      "bondOrder": [
        1,
      ],
      "bonds": [
        12,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 13,
      "properties": {},
      "serial": 13,
      "x": -11.829,
      "y": 4.236,
      "z": 9.194,
    },
    {
      "atom": "N",
      "bondOrder": [
        4,
        4,
        1,
      ],
      "bonds": [
        11,
        15,
        17,
      ],
      "elem": "N",
      "hetflag": true,
      "index": 14,
      "properties": {},
      "serial": 14,
      "x": -8.933,
      "y": 6.343,
      "z": 9.563,
    },
    {
      "atom": "N",
      "bondOrder": [
        4,
        4,
      ],
      "bonds": [
        12,
        14,
      ],
      "elem": "N",
      "hetflag": true,
      "index": 15,
      "properties": {},
      "serial": 15,
      "x": -10.075,
      "y": 5.83,
      "z": 10.001,
    },
    {
      "atom": "O",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        11,
        18,
      ],
      "elem": "O",
      "hetflag": true,
      "index": 16,
      "properties": {},
      "serial": 16,
      "x": -7.582,
      "y": 6.162,
      "z": 7.573,
    },
    {
      "atom": "H",
      "bondOrder": [
        1,
      ],
      "bonds": [
        14,
      ],
      "elem": "H",
      "hetflag": true,
      "index": 17,
      "properties": {},
      "serial": 17,
      "x": -8.359,
      "y": 6.978,
      "z": 10.05,
    },
    {
      "atom": "H",
      "bondOrder": [
        1,
      ],
      "bonds": [
        16,
      ],
      "elem": "H",
      "hetflag": true,
      "index": 18,
      "properties": {},
      "serial": 18,
      "x": -7.471,
      "y": 5.782,
      "z": 6.724,
    },
  ],
  [],
]
`;

exports[`Function SDF | parseV2000 | options: multimodel, onemol | Atoms should match the snapshot when multimodel is true and onemol is true 1`] = `
[
  [
    {
      "atom": "C",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        4,
        10,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 0,
      "properties": {},
      "serial": 0,
      "x": -9.856,
      "y": 4.168,
      "z": 6.715,
    },
    {
      "atom": "C",
      "bondOrder": [
        1,
      ],
      "bonds": [
        2,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 1,
      "properties": {},
      "serial": 1,
      "x": -12.153,
      "y": 2.39,
      "z": 5.307,
    },
    {
      "atom": "C",
      "bondOrder": [
        1,
        4,
        4,
      ],
      "bonds": [
        1,
        4,
        5,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 2,
      "properties": {},
      "serial": 2,
      "x": -11.98,
      "y": 3.867,
      "z": 5.218,
    },
    {
      "atom": "C",
      "bondOrder": [
        4,
        4,
        1,
      ],
      "bonds": [
        4,
        6,
        7,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 3,
      "properties": {},
      "serial": 3,
      "x": -11.355,
      "y": 5.923,
      "z": 5.494,
    },
    {
      "atom": "C",
      "bondOrder": [
        1,
        4,
        4,
      ],
      "bonds": [
        0,
        2,
        3,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 4,
      "properties": {},
      "serial": 4,
      "x": -10.967,
      "y": 4.626,
      "z": 5.788,
    },
    {
      "atom": "N",
      "bondOrder": [
        4,
        4,
      ],
      "bonds": [
        2,
        6,
      ],
      "elem": "N",
      "hetflag": true,
      "index": 5,
      "properties": {},
      "serial": 5,
      "x": -12.865,
      "y": 4.669,
      "z": 4.596,
    },
    {
      "atom": "N",
      "bondOrder": [
        4,
        4,
        1,
      ],
      "bonds": [
        3,
        5,
        8,
      ],
      "elem": "N",
      "hetflag": true,
      "index": 6,
      "properties": {},
      "serial": 6,
      "x": -12.483,
      "y": 5.926,
      "z": 4.762,
    },
    {
      "atom": "O",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        3,
        9,
      ],
      "elem": "O",
      "hetflag": true,
      "index": 7,
      "properties": {},
      "serial": 7,
      "x": -10.714,
      "y": 7.084,
      "z": 5.905,
    },
    {
      "atom": "H",
      "bondOrder": [
        1,
      ],
      "bonds": [
        6,
      ],
      "elem": "H",
      "hetflag": true,
      "index": 8,
      "properties": {},
      "serial": 8,
      "x": -12.943,
      "y": 6.722,
      "z": 4.411,
    },
    {
      "atom": "H",
      "bondOrder": [
        1,
      ],
      "bonds": [
        7,
      ],
      "elem": "H",
      "hetflag": true,
      "index": 9,
      "properties": {},
      "serial": 9,
      "x": -9.929,
      "y": 7.031,
      "z": 6.415,
    },
    {
      "atom": "C",
      "bondOrder": [
        1,
        4,
        4,
      ],
      "bonds": [
        0,
        11,
        12,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 10,
      "properties": {},
      "serial": 10,
      "x": -9.716,
      "y": 4.99,
      "z": 7.966,
    },
    {
      "atom": "C",
      "bondOrder": [
        4,
        4,
        1,
      ],
      "bonds": [
        10,
        14,
        16,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 11,
      "properties": {},
      "serial": 11,
      "x": -8.698,
      "y": 5.843,
      "z": 8.337,
    },
    {
      "atom": "C",
      "bondOrder": [
        4,
        1,
        4,
      ],
      "bonds": [
        10,
        13,
        15,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 12,
      "properties": {},
      "serial": 12,
      "x": -10.564,
      "y": 5.007,
      "z": 9.052,
    },
    {
      "atom": "C",
      "bondOrder": [
        1,
      ],
      "bonds": [
        12,
      ],
      "elem": "C",
      "hetflag": true,
      "index": 13,
      "properties": {},
      "serial": 13,
      "x": -11.829,
      "y": 4.236,
      "z": 9.194,
    },
    {
      "atom": "N",
      "bondOrder": [
        4,
        4,
        1,
      ],
      "bonds": [
        11,
        15,
        17,
      ],
      "elem": "N",
      "hetflag": true,
      "index": 14,
      "properties": {},
      "serial": 14,
      "x": -8.933,
      "y": 6.343,
      "z": 9.563,
    },
    {
      "atom": "N",
      "bondOrder": [
        4,
        4,
      ],
      "bonds": [
        12,
        14,
      ],
      "elem": "N",
      "hetflag": true,
      "index": 15,
      "properties": {},
      "serial": 15,
      "x": -10.075,
      "y": 5.83,
      "z": 10.001,
    },
    {
      "atom": "O",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        11,
        18,
      ],
      "elem": "O",
      "hetflag": true,
      "index": 16,
      "properties": {},
      "serial": 16,
      "x": -7.582,
      "y": 6.162,
      "z": 7.573,
    },
    {
      "atom": "H",
      "bondOrder": [
        1,
      ],
      "bonds": [
        14,
      ],
      "elem": "H",
      "hetflag": true,
      "index": 17,
      "properties": {},
      "serial": 17,
      "x": -8.359,
      "y": 6.978,
      "z": 10.05,
    },
    {
      "atom": "H",
      "bondOrder": [
        1,
      ],
      "bonds": [
        16,
      ],
      "elem": "H",
      "hetflag": true,
      "index": 18,
      "properties": {},
      "serial": 18,
      "x": -7.471,
      "y": 5.782,
      "z": 6.724,
    },
  ],
]
`;
